server/pulp/server/db/querysets.py

```diff
@@ -67,6 +67,19 @@ def update(self, *args, **kwargs):
                 # if post_save() is not defined for this particular document, that's ok
                 pass
 
+    def get_or_404(self, **kwargs):
```

Comment:
This is based on a Django shortcut function. 
https://docs.djangoproject.com/en/1.8/topics/http/shortcuts/#get-object-or-404

Since we handle 404's by raising an exception, this replaces code that is repeated all over the code base:

    thing = model.Collection.objects().first()
    if thing is None:
        raise MissingResource

This does more than simply reduce lines of code. Everywhere that this is used, the function has one fewer code path that needs to be tested. I agree that it is a little weird, but I think that the benefits outweigh the awkwardness. Thoughts?